backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/TypeOperatorLowering.kt

```diff
@@ -75,6 +63,34 @@ private class TypeOperatorTransformer(val context: CommonBackendContext, val fun
         }
     }
 
+    fun IrBuilderWithScope.irCast(argument: IrExpression, type: IrType, argumentIsNotNull: Boolean = false) = irBlock {
+        val value = irTemporary(argument)
+        +irIfThenElse(
+                type,
+                condition = irInstanceOf(value, type, argumentIsNotNull),
+                thenPart = irImplicitCast(irGet(value), type),
+                elsePart = irCall(symbols.ThrowTypeCastException)
```

Comment:
If I understand correctly, previous implementation of casts was throwing an exception with proper message in some cases. This implementation doesn't seem to do so.

If we don't have a test for the message, please add it.